### DISCUSSION TITLE
Deployed version is different from local dev version. Please help

### DISCUSSION THREAD

**Discussion by toughyear** [vercel/next.js#15012] [Help]
I am using a NPM package @brainhubeu/react-carousel, the site works fine in local dev environment.
The following is an image on my local machine working perfectly fine.
![image](https://user-images.githubusercontent.com/38691925/87044134-c09e7a00-c213-11ea-8a40-b00857110f57.png)

But when I deploy it on Vercel, the hosted site is quite different - 
https://curate.now.sh/ (live)
![image](https://user-images.githubusercontent.com/38691925/87044258-eb88ce00-c213-11ea-933a-618d9fac0d3e.png)

I initially thought it was due the fact that the library might not support for SSR so I used the following solution for importing the file- 
`import dynamic from "next/dynamic";

const Carousel = dynamic(() => import("@brainhubeu/react-carousel"), {
  ssr: false,
});
`

I am also importing the CSS files of the library in the_app.js but not using next/dynamic

Any pointers as to what might be going wrong?

**Comment by jamesmosier** [comment]
Looks like [they recommend importing it like this](https://github.com/brainhubeu/react-carousel#ssr):

```js
import dynamic from 'next/dynamic';

const { default: Carousel, Dots } = dynamic(
 () => require('@brainhubeu/react-carousel'),
 { ssr: false },
);
```

next/dynamic only works for importing React components themselves, not libraries entirely.

**Reply by toughyear** [reply]
I am getting a new error on my local machine:

Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.

Check the render method of `About`.


About component contains the Carousel component that was imported dynamically.

**Comment by toughyear** [comment]
Although when I import like this -

`const Carousel = dynamic(() => import("@brainhubeu/react-carousel"), {
  ssr: false,
});`

It works locally. but again not on the deployed site. To deploy I am using default NextJS settings. Do you think it should be an error from Vercel deployment settings?

It's CSS issue so I am not sure.



This carousel component is part of a component called About.
I am importing this component in my index.js page file as :
`import About from "../components/About";`

IT works locally but not deployed.

If I import it as - 

```import dynamic from "next/dynamic";

const About = dynamic(() => import("../components/About"));
```

It again works locally but not deployed.

**Reply by banstack** [reply]
I'm also having a similar issue did you ever figure it out @toughyear ?
